overlays/haskell.nix

```diff
@@ -11,7 +11,7 @@ final: prev: {
         # overlays.
         defaultModules = [];
 
-        # TODO: doc etc
+        # Additional user-provided mappings to augment ./../lib/pkgconf-nixpkgs-map.nix
         extraPkgconfigMappings = {};
```

Comment:
re: @L-as 

> This should probably be part of the project-common module.

What did you mean by this? I'm not familiar with the whole structure of haskell.nix, but would that also make it easy to set these options globally?

Reply:
@ramirez7 Sorry for the late response, yeah, essentially you add an option for it in `modules/project-common.nix`, then it's in the result of evaluating the module (see `cabalProject'`) .

Reply:
```nix
    extraPkgconfigMappings = mkOption {
      type = submodule [ fill this out ];
      default = {};
      description = ''
        Extra pkgconfig mappings
      '';
    };
```

Reply:
hm I'll try this in a branch and get a good comparison going